src/url-replacements.js

```diff
@@ -239,6 +242,14 @@ class UrlReplacements {
       return this.getTimingData_('navigationStart',
           'domContentLoadedEventStart');
     });
+
+    // Returns an identifier for the viewer.
+    this.set_('VIEWER', () => {
+      const viewerUrl = viewerFor(this.win_).getParam(VIEWPORT_ORIGIN_PARAM_) ||
```

Comment:
I had to translate from a (small) set of UserAgents into a referrer since some services do not provide a referrer.

But we need to keep the distinction between viewer referrers (maybe "viewer origin"?) vs referrer. When you open an AMP doc (let's call it "A") from the SERP viewer, both the viewer referrer and referrer should be "google.com". If we were to follow some link in that doc to another AMP doc (called "B", and the viewer could load it), the viewer referrer should still be "google.com" but the referrer should be doc A.